plugins/nf-azure/src/main/nextflow/cloud/azure/fusion/AzFusionEnv.groovy

```diff
@@ -41,6 +41,24 @@ class AzFusionEnv implements FusionEnv {
         if (scheme != 'az') {
             return Collections.<String, String> emptyMap()
         }
+        // Delegate to the 3-parameter version with poolIdentityClientId from batch config
+        final cfg = AzConfig.config
+        return getEnvironment(scheme, config, cfg.batch().poolIdentityClientId)
+    }
+
+    /**
+     * Get the environment variables for Fusion with Azure, taking into 
+     * account a managed identity ID if provided
+     *
+     * @param scheme The storage scheme ('az' for Azure)
+     * @param config The Fusion configuration
+     * @param managedIdentityId Optional managed identity client ID from pool options
+     * @return Map of environment variables
+     */
+    Map<String, String> getEnvironment(String scheme, FusionConfig config, String managedIdentityId) {
```

Comment:
Bit confused this version of `getEnvironment`, the `config` is parameter is not used, but internally is used `AzConfig.config`? Is that done on purpose?